### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException: When using RasterRDD call

### ISSUE THREAD

**Issue by owcm** [geotrellis/geotrellis#1304]
Using geotrellis-spark specifically 0.10.0-2283ec5, Was updating to the new Tiling System. 

Reading in Geotiffs that are 1500 x 1500 in size.  Reading in 1 file works fine.  Reading 1+ files causes the exception.  Stack trace listed below.

Code is inside a spark context runnable.
```
**Calling code**
    // Read in 4  geotiff tiles return as  an RDD from S3
    val inputTiles = BuildRdd.tilePrepRoute( sc,  4 )

    println( "Return  BuildRdd " )

    // Use the "RasterMetaData.fromRdd" call to find the zoom
    // level that the closest match to the resolution of our source image,
    // and derive information such as the full bounding box and data type.
    val (_, rasterMetaData) =
      RasterMetaData.fromRdd(inputTiles, FloatingLayoutScheme(512))

    println( "Return  RasterMetaData " )
    // Use the Tiler to cut our tiles into tiles that are index by the z/x/y map coordinates.
    val tiled: RDD[(SpatialKey, Tile)] = inputTiles.tileToLayout(rasterMetaData, Bilinear)

    println( "Return  inputTiles " )

    // We'll be tiling the images using a zoomed layout scheme
    // in the web mercator format (which fits the slippy map tile specification).
    // We'll be creating 256 x 256 tiles.
    val layoutScheme = ZoomedLayoutScheme(WebMercator, tileSize = 256)
    // We need to reproject the tiles to WebMercator
    println( "RasterRDD Start " )
    val (zoom, reprojected) = RasterRDD(tiled, rasterMetaData).reproject(WebMercator, layoutScheme, Bilinear)
  println( "RasterRDD End " )
 ==========================================================================

**Body of RDD Constructor**

 def tilePrepRoute( implicit sc: SparkContext, desiredFilesToProcess: Int) :RDD[ (ProjectedExtent, Tile) ] =  {

    val sourcePath = "va-1m"+ "/sources"
    var fileListAll = AmazonConnect.getObjectsInBucket(sourcePath)

    var tileList= new ListBuffer[(ProjectedExtent,Tile)]()

    var filesToProcess = 0
    if ( desiredFilesToProcess == -1 ) {
      filesToProcess = fileListAll.length
    } else {
      filesToProcess = desiredFilesToProcess
    }

    val fileList = fileListAll.take( filesToProcess )

    var fileCount = 0
    var totalTileSize = 0

    for (name <- fileList) {

      if ( name._2.toLowerCase.contains( ".tif") ) {

        val bytesToUse = AmazonConnect.getS3Object(name._2)

        val gtIn = SingleBandGeoTiff(bytesToUse)

        // Returns a ProjectedRaster
        val projectedRaster = gtIn.projectedRaster

        val ProjectedRaster(Raster(tile, extent), crs) = projectedRaster.reproject(WebMercator)

        // Returns SingleBandGeoTiff
        val reprojGeoTiff = SingleBandGeoTiff(tile, extent, crs, gtIn.tags, gtIn.options)

        println ( "Reprojected new extent: " + extent  )

        // Move from tile to rastersource
        val theRS = RasterSource( reprojGeoTiff, extent )

        // Write out slope product
        val lclTile:Tile = theRS.get
        val lclExtent = theRS.rasterExtent.get.extent
        val projExt = new ProjectedExtent( lclExtent, crs)
        val gtSlope = GeoTiff(lclTile, lclExtent, crs)

        val newEntry = (projExt, lclTile)
        tileList += newEntry

      }

    }
    val tiles = sc.parallelize( tileList )
    tiles

  }
```
=================================

Exception Stack 

java.lang.ArrayIndexOutOfBoundsException: 2253001
[error] 	at geotrellis.raster.FloatArrayTile.updateDouble(FloatArrayTile.scala:33)
[error] 	at geotrellis.raster.MutableArrayTile$class.setDouble(MutableArrayTile.scala:37)
[error] 	at geotrellis.raster.FloatArrayTile.setDouble(FloatArrayTile.scala:27)
[error] 	at geotrellis.raster.MutableArrayTile$class.update(MutableArrayTile.scala:43)
[error] 	at geotrellis.raster.FloatArrayTile.update(FloatArrayTile.scala:27)
[error] 	at geotrellis.raster.stitch.Stitcher$TileStitcher$$anonfun$stitch$2.apply(Stitcher.scala:14)
[error] 	at geotrellis.raster.stitch.Stitcher$TileStitcher$$anonfun$stitch$2.apply(Stitcher.scala:13)
[error] 	at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:772)
[error] 	at scala.collection.immutable.List.foreach(List.scala:318)
[error] 	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:771)
[error] 	at geotrellis.raster.stitch.Stitcher$TileStitcher$.stitch(Stitcher.scala:13)
[error] 	at geotrellis.raster.stitch.Stitcher$TileStitcher$.stitch(Stitcher.scala:10)
[error] 	at geotrellis.spark.buffer.BufferTiles$$anonfun$bufferWithNeighbors$1$$anonfun$apply$2.apply(BufferTiles.scala:118)
[error] 	at geotrellis.spark.buffer.BufferTiles$$anonfun$bufferWithNeighbors$1$$anonfun$apply$2.apply(BufferTiles.scala:83)
[error] 	at scala.Option.map(Option.scala:145)
[error] 	at geotrellis.spark.buffer.BufferTiles$$anonfun$bufferWithNeighbors$1.apply(BufferTiles.scala:83)
[error] 	at geotrellis.spark.buffer.BufferTiles$$anonfun$bufferWithNeighbors$1.apply(BufferTiles.scala:82)
[error] 	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$flatMapValues$1$$anonfun$apply$43$$anonfun$apply$44.apply(PairRDDFunctions.scala:712)
[error] 	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$flatMapValues$1$$anonfun$apply$43$$anonfun$apply$44.apply(PairRDDFunctions.scala:711)
[error] 	at scala.collection.Iterator$$anon$13.hasNext(Iterator.scala:371)
[error] 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
[error] 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
[error] 	at org.apache.spark.rdd.RDD$$anonfun$reduce$1$$anonfun$14.apply(RDD.scala:992)
[error] 	at org.apache.spark.rdd.RDD$$anonfun$reduce$1$$anonfun$14.apply(RDD.scala:991)
[error] 	at org.apache.spark.SparkContext$$anonfun$36.apply(SparkContext.scala:1943)
[error] 	at org.apache.spark.SparkContext$$anonfun$36.apply(SparkContext.scala:1943)
[error] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:66)
[error] 	at org.apache.spark.scheduler.Task.run(Task.scala:88)
[error] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:214)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[error] 	at java.lang.Thread.run(Thread.java:745)
[error] 16/01/29 17:08:55 WARN TaskSetManager: Lost task 4.0 in stage 9.0 (TID 60, localhost): java.lang.ArrayIndexOutOfBoundsException: 2253001
[error] 	at geotrellis.raster.FloatArrayTile.updateDouble(FloatArrayTile.scala:33)
[error] 	at geotrellis.raster.MutableArrayTile$class.setDouble(MutableArrayTile.scala:37)
[error] 	at geotrellis.raster.FloatArrayTile.setDouble(FloatArrayTile.scala:27)
[error] 	at geotrellis.raster.MutableArrayTile$class.update(MutableArrayTile.scala:43)
[error] 	at geotrellis.raster.FloatArrayTile.update(FloatArrayTile.scala:27)
[error] 	at geotrellis.raster.stitch.Stitcher$TileStitcher$$anonfun$stitch$2.apply(Stitcher.scala:14)
[error] 	at geotrellis.raster.stitch.Stitcher$TileStitcher$$anonfun$stitch$2.apply(Stitcher.scala:13)
[error] 	at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:772)
[error] 	at scala.collection.immutable.List.foreach(List.scala:318)
[error] 	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:771)
[error] 	at geotrellis.raster.stitch.Stitcher$TileStitcher$.stitch(Stitcher.scala:13)
[error] 	at geotrellis.raster.stitch.Stitcher$TileStitcher$.stitch(Stitcher.scala:10)
[error] 	at geotrellis.spark.buffer.BufferTiles$$anonfun$bufferWithNeighbors$1$$anonfun$apply$2.apply(BufferTiles.scala:118)
[error] 	at geotrellis.spark.buffer.BufferTiles$$anonfun$bufferWithNeighbors$1$$anonfun$apply$2.apply(BufferTiles.scala:83)
[error] 	at scala.Option.map(Option.scala:145)
[error] 	at geotrellis.spark.buffer.BufferTiles$$anonfun$bufferWithNeighbors$1.apply(BufferTiles.scala:83)
[error] 	at geotrellis.spark.buffer.BufferTiles$$anonfun$bufferWithNeighbors$1.apply(BufferTiles.scala:82)
[error] 	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$flatMapValues$1$$anonfun$apply$43$$anonfun$apply$44.apply(PairRDDFunctions.scala:712)
[error] 	at org.apache.spark.rdd.PairRDDFunctions$$anonfun$flatMapValues$1$$anonfun$apply$43$$anonfun$apply$44.apply(PairRDDFunctions.scala:711)
[error] 	at scala.collection.Iterator$$anon$13.hasNext(Iterator.scala:371)
[error] 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
[error] 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
[error] 	at org.apache.spark.rdd.RDD$$anonfun$reduce$1$$anonfun$14.apply(RDD.scala:992)
[error] 	at org.apache.spark.rdd.RDD$$anonfun$reduce$1$$anonfun$14.apply(RDD.scala:991)
[error] 	at org.apache.spark.SparkContext$$anonfun$36.apply(SparkContext.scala:1943)
[error] 	at org.apache.spark.SparkContext$$anonfun$36.apply(SparkContext.scala:1943)
[error] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:66)
[error] 	at org.apache.spark.scheduler.Task.run(Task.scala:88)
[error] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:214)
[error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[error] 	at java.lang.Thread.run(Thread.java:745)

**Comment by owcm** [comment]
Update to :  "com.azavea.geotrellis" %% "geotrellis-spark" % "0.10.0-2283ec5",
  "com.azavea.geotrellis" %% "geotrellis-raster" % "0.10.0-2283ec5",
  "com.azavea.geotrellis" %% "geotrellis-engine" % "0.10.0-2283ec5",

and still receiving the some stack error.

**Comment by lossyrob** [comment]
@owcm that's still an older version than the fix.

Can you try with `0.10.0-82f0170`?

**Comment by owcm** [comment]
Adding link to zip with project and data files with which the problem can be recreated.

https://drive.google.com/file/d/0B1YZbN2gCsmSVV9ZaE9mUjNBeEE/view?usp=sharing

This is is modified/hacked version of the geotrellis/geotrellis-landsat-tutorial project. 

In file BuildRdd.scala I have 2 filelist code blocks.  One list works the other list causes the failures. The failed file list is currently commented out.

Let me know if you need more.

**Comment by owcm** [comment]
@lossyrob My apologies I missed your comment about trying 0.10.0-82f0170. Tried up grading today.  Looks like packages have moved around a bit.  I had to make some code changes:

Colorbreaks no longer seem to be supported.  getting warning

**Comment by owcm** [comment]
Okay figured out that  KryoSerializer has moved to "geotrellis.spark.io.kryo.KryoRegistrator.

So know I can read in the files and tile them with out error.  

I cannot visualize my result yet because I have not figured out how to replace colorbreaks.

But I think you can say that this issue is fixed under  0.10.0-82f0170.

**Comment by owcm** [comment]
Replaced colorbreaks with StrictColorClassifier so know I can visualize result

**Comment by moradology** [comment]
@owcm The documentation for some API changes is just starting to really come together - sorry for any frustration that might have been caused by the lack of advertisement/obviousness there. FWIW, here are some docs on the use of ColorClassifiers: https://github.com/geotrellis/geotrellis/blob/master/docs/raster/rendering.md#color-classification

**Comment by lossyrob** [comment]
@owcm awesome, thank you! Closing this issue as fixed by #1352
